runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/validator/AvroValidator.java

```diff
@@ -33,13 +32,17 @@
 import io.aklivity.zilla.runtime.binding.kafka.internal.config.KafkaCatalogConfig;
 import io.aklivity.zilla.runtime.binding.kafka.internal.validator.config.AvroValidatorConfig;
 import io.aklivity.zilla.runtime.engine.catalog.CatalogHandler;
+import io.aklivity.zilla.runtime.engine.catalog.ParsedSchema;
 
 public final class AvroValidator implements Validator
 {
     private static final byte MAGIC_BYTE = 0x0;
 
     private final List<KafkaCatalogConfig> catalogs;
+    private final KafkaCatalogConfig catalog;
     private final Long2ObjectHashMap<CatalogHandler> handlersById;
+    private final String keySubject;
+    private final String valueSubject;
```

Comment:
When we associate the validator via kafka binding options, we do so in the context of the key or the value, not both at once.

Shouldn't we have a single subject here, populated with the appropriate name, then no need for `isKey` in any validator method signature to split it back apart again?

Remember that the validator abstraction is generic, so it should not have any Kafka-specific concepts like `key` in the method signatures, as that doesn't make sense for other non-kafka protocol bindings.